Extensions/one_click_postage.js

```diff
@@ -376,63 +366,56 @@ XKit.extensions.one_click_postage = new Object({
 		}
 		m_object.custom_tweet = "";
 
-		XKit.interface.kitty.get(function(kitty_data) {
+		XKit.interface.kitty.get(kitty_data => {
 			if (kitty_data.errors === true) {
 				// We fucked up for some reason.
 				if (retry_mode !== true) {
-					XKit.extensions.one_click_postage.in_blog_process(data, state, obj, m_object, true);
+					this.in_blog_process(data, state, obj, m_object, true);
 				} else {
-					XKit.extensions.one_click_postage.show_error("INOCP109-SFORMKEYFAIL", state);
+					this.show_error("INOCP109-SFORMKEYFAIL", state);
 					$(obj).removeClass("xkit-button-working");
 					$(obj).addClass("xkit-button-error");
 				}
 				return;
 			}
 
-			XKit.tools.Nx_XHR({
-				method: "POST",
-				url: "https://www.tumblr.com/svc/post/update",
-				data: JSON.stringify(m_object),
-				json: true,
-				headers: {
-					"X-tumblr-puppies": kitty_data.kitten,
-					"X-tumblr-form-key": XKit.interface.form_key(),
-				},
-				onerror: function(response) {
-					XKit.interface.kitty.set("");
-					if (response.status === 401) {
-						XKit.extensions.one_click_postage.show_error("INOCP101");
+			XKit.svc.post.update(m_object, kitty_data.kitten)
+			.then(response => {
+				try {
+					let responseData = response.json();
+					if (responseData.errors === false) {
```

Comment:
Shouldn't XKit.svc.post functions just throw errors for `response.json()` and `responseData.errors === true`? then we can put all of our error handling logic into a catch handler.

Reply:
honestly I'm not sure why either of these things are necessary to check anyway, surely only non-OK responses would either have `errors` or a non-JSON response? i didnt really question it, I was just rewriting the same logic

Reply:
im still not sure what to do here; i'm not trying to do anything except refactor the requests into per-url promises, so trying to restructure how the requests are done wrt `onload`/`onerror` here is gonna be a ton more work that could be done later instead

Reply:
I can't see how this can be done separately from the other point/comment so I'm gonna resolve this conversation too. would absolutely love to do this in a followup PR though